betty/cache/file.py

```diff
@@ -0,0 +1,177 @@
+"""
+Provide caching that persists cache items to files.
+"""
+from __future__ import annotations
+
+import asyncio
+import shutil
+from collections.abc import Sequence
+from contextlib import suppress
+from os import utime
+from pathlib import Path
+from pickle import dumps, loads
+from typing import Generic, Any, Self
+
+import aiofiles
+from aiofiles.ospath import getmtime
+
+from betty.cache import CacheItem, CacheItemValueT
+from betty.cache._base import _CommonCacheBase
+from betty.locale import Localizer
+
+
+class _FileCacheItem(CacheItem[CacheItemValueT], Generic[CacheItemValueT]):
+    __slots__ = '_id', '_cache', '_modified'
+
+    def __init__(
+        self,
+        id: str,
+        cache: _FileCache[Any],
+        modified: int | float,
+    ):
+        self._id = id
+        self._cache = cache
```

Comment:
Keeping the cache inside items makes it easy to introduce cyclic dependencies. Remove this?